nixos/modules/services/misc/apache-kafka.nix

```
@@ -112,27 +164,49 @@ in {
       defaultText = literalExpression "pkgs.apacheKafka.passthru.jre";
       type = types.package;
     };
-
   };
 
-  config = mkIf cfg.enable {
-
-    environment.systemPackages = [cfg.package];
+  imports = [
+    (mkRenamedOptionModule
+      [ "services" "apache-kafka" "brokerId" ]
+      [ "services" "apache-kafka" "settings" ''"broker.id"'' ])
```

This escaping shouldn't be necessary.

```suggestion
      [ "services" "apache-kafka" "settings" "broker.id" ])
```

If anything tries to parse that `.` or renders this as an option path without escaping, that's its mistake, not a mistake here.

---

Understood, removed :)